examples/external-data/src/mock-customer-service/service.ts

```diff
@@ -229,6 +273,64 @@ export async function initializeCustomerService(props: ServiceProps): Promise<Se
 		result.send();
 	});
 
+	/**
+	 * An 'events' endpoint that can be called by Fluid services.
+	 *
+	 * For the 'session-end' event {@link SessionEndEventsListenerRequest}: If, after unregistering the given client url, there are any task ids that have an outstanding
+	 * webhook registered using this services internal '/external-data-webhook' endpoint but has no respective active client sessions mapped to it anymore,
+	 * then that webhook will be deregistered.
+	 *
+	 * @remarks Currently, the only supported request type is 'session-end' {@link SessionEndEventsListenerRequest} which enables the Fluid service to notify this service
+	 * that a particular Fluid session has ended which in turn causes this service to unregister any related webhooks to the respective Fluid session.
+	 */
+	expressApp.post("/events-listener", (request: EventsListenerRequest, result) => {
+		const eventType = request.body?.type;
+
+		if (eventType === "session-end") {
+			const typedRequest = request as SessionEndEventsListenerRequest;
```

Comment:
Nit: we can implement a helper method `static function isThis(obj: unknown) is SessionEndEventsListenerRequest` on `SessionEndEventsListenerRequest` that does exactly this check but saves us from having to then cast here and still maintain type safety.

Reply:
I'd prefer to make this part of a larger effort to have consistent request object validation and typecasting across all endpoints rather than have it one off here.

Also we should consider whether that adds unnecessary complexity to an example application